docs(en): add runtime config (#890)
nuxt/nuxtjs.org: content/en/guides/directory-structure/nuxt-config.md

</base-alert>

> ### env

Reviewer: At least current version of nuxt didn't deprecate `env` option and it has it's purpose of inlining environment variables (We also auto add `NUXT_ENV_*` vars).

I think it worth to be kept in docs with some explanation about use-case and difference (not sure if any other page exists regarding `env` option or not)

Author: ok improved this @pi0 let me know if it is better or you want to add further explantion

Reviewer: Seems awesome 👍

- is available using `$config` in both server and client.

```js{}[nuxt.config.js]
export default {
  publicRuntimeConfig: {
    baseURL: process.env.BASE_URL || 'https://nuxtjs.org'
  }
}
```

#### `privateRuntimeConfig`

- should hold all env variables that are private and that should not be exposed on the frontend. This could include a reference to your API secret tokens for example.
- is only available on server using same `$config` (it overrides publicRuntimeConfig)

```js{}[nuxt.config.js]
export default {
  privateRuntimeConfig: {
    apiSecret: process.env.API_SECRET
  }
}
```

#### **Using your config values:**

You can then access these values anywhere by using the context in your pages, store, components and plugins by using `this.$config` or `context.$config`.

```html{}[pages/index.vue]
<script>
  asyncData ({ $config: { baseURL } }) {
    const posts = await fetch(`${baseURL}/posts`)
      .then(res => res.json())
  }
</script>
```

Inside your templates you can access your runtimeConfigs directly using `$config.*`

```html{}[pages/index.vue]
<template>
  <p>Our Url is: {{ $config.baseURL}}</p>
</template>
```

<base-alert type="warn">

> You can expose server/private tokens if using `$config` in fetch/asyncData or directly inside your template.

Reviewer: ```suggestion
It is possible that private config to be exposed to client when using `$config` in `fetch`, `asyncData` or directly inside templates. 
```

/cc @danielroe @manniL  if this can be reworded better

Reviewer 2: ```suggestion
Your private config could be exposed if you use `$config` outside of a server-only context (for example, if you use `$config` in `fetch`, `asyncData` or directly inside your template).
```

</base-alert>

### runtimeConfig

#### `publicRuntimeConfig`

- should hold all env variables that are public as these will be exposed on the frontend. This could include a reference to your public URL for example.
- is available using `$config` in both server and client.

```js{}[nuxt.config.js]
export default {
  publicRuntimeConfig: {
    baseURL: process.env.BASE_URL || 'https://nuxtjs.org'
  }
}
```

#### `privateRuntimeConfig`

- should hold all env variables that are private and that should not be exposed on the frontend. This could include a reference to your API secret tokens for example.
- is only available on server using same `$config` (it overrides publicRuntimeConfig)

```js{}[nuxt.config.js]
export default {
  privateRuntimeConfig: {
    apiSecret: process.env.API_SECRET
  }
}
```

#### **Using your config values:**

You can then access these values anywhere by using the context in your pages, store, components and plugins by using `this.$config` or `context.$config`.

```html{}[pages/index.vue]
<script>
  asyncData ({ $config: { baseURL } }) {
    const posts = await fetch(`${baseURL}/posts`)
      .then(res => res.json())
  }
</script>
```

> Inside your templates you can access your runtimeConfigs directly using `$config.*`

Reviewer: Maybe worth mentioning that regardless, if rendering `$config.token` it will be displayed in page (someone had this question in one issue)

Author: warning added .let me know if it needs further explanation
